### PR TITLE
Updating the service provider to work with SDKv3 and Laravel 5.1

### PR DESCRIPTION
**Pull request by jeremeamia**
Feedback welcome.

### CODE REVIEW

**Comment by jeremeamia** [comment]
@GrahamCampbell CR, please?

File: composer.json [hunk 1]
```diff
@@ -12,15 +12,16 @@
         }
     ],
     "require": {
-        "php": ">=5.4.0",
-        "aws/aws-sdk-php": "~2.4",
-        "illuminate/support": "~5.0"
+        "php": ">=5.5.0",
```
Comment (GrahamCampbell) [inline composer.json]: Better to make that 5.5.9.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: php
 
 php:
-  - 5.4
```
Comment (GrahamCampbell) [inline .travis.yml]: Better to change this to 5.5.9, so we'r testing on the lowest supported version too.

File: composer.json [hunk 3]
```diff
@@ -12,15 +12,16 @@
...
     "require-dev": {
-        "laravel/framework": "~5.0",
+        "laravel/framework": "~5.1",
         "phpunit/phpunit": "~4.0"
     },
     "autoload": {
         "psr-4": { "Aws\\Laravel\\": "src/" }
-    }
+    },
+    "minimum-stability": "dev"
```
Comment (GrahamCampbell) [inline composer.json]: Don't need this, just add `@dev` to the laravel version constraint. :)

**Comment by damiencriado** [comment]
Gogogo please

File: composer.json [hunk 4]
```diff
@@ -12,15 +12,18 @@
         }
     ],
     "require": {
-        "php": ">=5.4.0",
-        "aws/aws-sdk-php": "~2.4",
-        "illuminate/support": "~5.0"
+        "php": ">=5.5.9",
+        "aws/aws-sdk-php": "~3.0",
+        "illuminate/support": "~5.1"
```
Comment (GrahamCampbell) [inline composer.json]: Better to make this `5.1.*`.

Comment (GrahamCampbell) [inline composer.json]: Given that `~5.0` was already wrong before.

Comment (jeremeamia) [inline composer.json]: Laravel is not likely to change any parts that are relevant for a while. SDKv3 is the reason we had to bump to 5., so I think ~5.1 will be fine. If it ever becomes not fine, then we'll worry about it at that time.

**Comment by GrahamCampbell** [comment]
But it's too late then, because composer can still resolve the older versions of this package

**Comment by jeremeamia** [comment]
Not quite sure I follow. Let's open a new PR/issue to talk about though. I've already merged this to master. The 2.0 branch has the old codebase now.

**Pull request merged**